packages/block-library/src/columns/editor.scss

```diff
@@ -89,30 +89,24 @@
 
 			// Beyond mobile, allow 2 columns.
 			@include break-small() {
-				flex-basis: 50%;
+				flex-basis: calc(50% - #{$grid-size-large} - #{$block-padding * 1.5});
 				flex-grow: 0;
 			}
 
 			// Add space between columns. Themes can customize this if they wish to work differently.
 			// This has to match the same padding applied in style.scss.
 			// Only apply this beyond the mobile breakpoint, as there's only a single column on mobile.
 			@include break-small() {
-				&:nth-child(odd) {
-					margin-right: $grid-size-large * 2;
-				}
 				&:nth-child(even) {
 					margin-left: $grid-size-large * 2;
```

Comment:
Column gutters are too narrow in editor view compared to frontend. I think you need to do `(2 * $grid-size-large) + (2 * $block-padding)` here to account for the negative margins added by the editor. See #12816.

![image](https://user-images.githubusercontent.com/1050936/51936314-85b4d780-23cd-11e9-9831-f20d941e1abf.png)